spec/controllers/concerns/render_condition_concern_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'rails_helper'
+
+RSpec.describe RenderConditionConcern, type: :controller do
+  let(:all_feature?) { false }
+  let(:show_feature?) { false }
+
+  controller ApplicationController do
+    include RenderConditionConcern
+
+    render_if -> { FeatureManagement.all_feature? }
+    render_if -> { FeatureManagement.show_feature? }, only: [:show]
+
+    def index
+      render plain: ''
+    end
+
+    def show
+      render plain: ''
+    end
+  end
+
+  before do
+    allow(FeatureManagement).to receive(:all_feature?).and_return(all_feature?)
+    allow(FeatureManagement).to receive(:show_feature?).and_return(show_feature?)
+
+    routes.draw do
+      get 'index' => 'anonymous#index'
+      get 'show' => 'anonymous#show'
+    end
```

Comment:
do we need this? I think anonymous controllers "just work" without it usually

Reply:
It works by default for `index`, but not for `show`. Not sure why, tbh 🤷 

```
  1) RenderConditionConcern#show renders 404
     Failure/Error: subject(:response) { get :show }
     
     ActionController::UrlGenerationError:
       No route matches {:action=>"show", :controller=>"anonymous"}
     # ./spec/controllers/concerns/render_condition_concern_spec.rb:52:in `block (3 levels) in <top (required)>'
     # ./spec/controllers/concerns/render_condition_concern_spec.rb:55:in `block (3 levels) in <top (required)>'
```

Reply:
huh, weird :/

Reply:
A bit of light sleuthing, maybe `index` is available since it's set up as the default action handler.

- https://github.com/rspec/rspec-rails/blob/dc0e2b5/lib/rspec/rails/example/controller_example_group.rb#L94-L101
- https://github.com/rails/rails/blob/51b4370/actionpack/lib/action_dispatch/routing/mapper.rb#L157-L159